// reshow-flux/build/cjs/connect.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("reshow-runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("reshow-runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("reshow-runtime/helpers/possibleConstructorReturn"));

var _assertThisInitialized2 = _interopRequireDefault(require("reshow-runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("reshow-runtime/helpers/inherits"));

var _getPrototypeOf2 = _interopRequireDefault(require("reshow-runtime/helpers/getPrototypeOf"));

var _get2 = _interopRequireDefault(require("reshow-runtime/helpers/get"));

var _defineProperty2 = _interopRequireDefault(require("reshow-runtime/helpers/defineProperty"));

var _objectSpread2 = _interopRequireDefault(require("reshow-runtime/helpers/objectSpread2"));

var _array = _interopRequireDefault(require("array.dedup"));

var _reshowFluxBase = require("reshow-flux-base");

var DEFAULT_OPTIONS = {
  withProps: false
};
var keys = Object.keys;

var getProps = function getProps(props, opt) {
  return opt.withProps && props ? props : {};
};

var getState = function getState(self, prevState, maybeProps, opt) {
  return self.calculateState(prevState, getProps(maybeProps, opt));
};

var getStores = function getStores(self, maybeProps, opt) {
  return self.getStores(getProps(maybeProps, opt));
};

var connect = function connect(Base, options) {
  var thisOptions = (0, _objectSpread2["default"])({}, DEFAULT_OPTIONS, {}, options || {});

  var ConnectedClass =
  /*#__PURE__*/
  function (_Base) {
    (0, _inherits2["default"])(ConnectedClass, _Base);

    function ConnectedClass(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, ConnectedClass);
      _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(ConnectedClass).call(this, props));
      (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "__stores", []);
      (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "__handleChange", function () {
        if (!_this.__stores) {
          // avoid race condition
          return;
        }

        var con = _this.constructor;

        _this.setState(function (prevState, currentProps) {
          return getState(con, prevState, currentProps, thisOptions);
        });
      });
      (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "__setStores", function (stores) {
        if (_this.__stores) {
          _this.__resetStores();
        }

        stores = (0, _array["default"])(stores);
        (stores || []).forEach(function (store) {
          return store.addListener(_this.__handleChange, _reshowFluxBase.CHANGE);
        });
        _this.__stores = stores;
      });
      (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "__resetStores", function () {
        if (!_this.__stores) {
          return;
        }

        _this.__stores.forEach(function (store) {
          return store.removeListener(_this.__handleChange, _reshowFluxBase.CHANGE);
        });

        _this.__stores = null;
      });
      var _con = _this.constructor;

      if (!_con.calculateState) {
        _con.calculateState = (0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "calculateState", (0, _assertThisInitialized2["default"])(_this));
      }

      if (!_con.getStores) {
        _con.getStores = (0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "getStores", (0, _assertThisInitialized2["default"])(_this));
      }

      _con.calculateState.bind(_con);

      _con.getStores.bind(_con);

      if (props.withConstructor) {
        _this.__setStores(getStores(_con, props, thisOptions));
      }

      if (!_this.state) {
        _this.state = {};
      }

      if (!thisOptions.withProps) {
        var calculatedState = getState(_con, undefined, props, thisOptions);

        if (calculatedState) {
          keys(calculatedState).forEach(function (key) {
            return _this.state[key] = calculatedState[key];
          });
        }
      }

      return _this;
    }

    (0, _createClass2["default"])(ConnectedClass, [{
      key: "componentDidMount",
      value: function componentDidMount() {
        if ((0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "componentDidMount", this)) {
          (0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "componentDidMount", this).call(this);
        }

        if (this.props && !this.props.withConstructor) {
          this.__setStores(getStores(this.constructor, this.props, thisOptions));
        }
      }
    }, {
      key: "componentDidUpdate",
      value: function componentDidUpdate(prevProps, prevState) {
        if ((0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "componentDidUpdate", this)) {
          (0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "componentDidUpdate", this).call(this, prevProps, prevState);
        }

        if (thisOptions.withProps) {
          this.__setStores(getStores(this.constructor, this.props, thisOptions));
        }
      }
    }, {
      key: "componentWillUnmount",
      value: function componentWillUnmount() {
        if ((0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "componentWillUnmount", this)) {
          (0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass.prototype), "componentWillUnmount", this).call(this);
        }

        this.__resetStores();
      }
    }], [{
      key: "getDerivedStateFromProps",
      value: function getDerivedStateFromProps(nextProps, prevState) {
        var thisStates = null;

        if ((0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass), "getDerivedStateFromProps", this)) {
          thisStates = (0, _get2["default"])((0, _getPrototypeOf2["default"])(ConnectedClass), "getDerivedStateFromProps", this).call(this, nextProps, prevState);
        }

        if (thisOptions.withProps) {
          var calState = getState(ConnectedClass, (0, _objectSpread2["default"])({}, prevState, {}, thisStates), nextProps, thisOptions);
          thisStates = (0, _objectSpread2["default"])({}, thisStates, {}, calState);
        }

        return thisStates;
      }
    }]);
    return ConnectedClass;
  }(Base);

  var componentName = Base.displayName || Base.name;
  ConnectedClass.displayName = 'FluxConnected(' + componentName + ')';
  return ConnectedClass;
};

var _default = connect;
exports["default"] = _default;
module.exports = exports.default;
