// organism-react-scroll-nav/build/cjs/ui/organisms/ScrollReceiver.js
"use strict";

var _interopRequireWildcard = require("reshow-runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("reshow-runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("reshow-runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("reshow-runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("reshow-runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("reshow-runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("reshow-runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("reshow-runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _reshowFlux = require("reshow-flux");

var _reactAtomicMolecule = require("react-atomic-molecule");

var _scrollStore = _interopRequireDefault(require("../../src/stores/scrollStore"));

var _fastScrollStore = _interopRequireDefault(require("../../src/stores/fastScrollStore"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var ScrollReceiver =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2["default"])(ScrollReceiver, _Component);

  function ScrollReceiver() {
    (0, _classCallCheck2["default"])(this, ScrollReceiver);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(ScrollReceiver).apply(this, arguments));
  }

  (0, _createClass2["default"])(ScrollReceiver, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          atom = _this$props.atom,
          container = _this$props.container,
          scrollMargin = _this$props.scrollMargin,
          targetId = _this$props.targetId,
          isScrollReceiver = _this$props.isScrollReceiver,
          noDelay = _this$props.noDelay,
          children = _this$props.children,
          resetProps = (0, _objectWithoutProperties2["default"])(_this$props, ["atom", "container", "scrollMargin", "targetId", "isScrollReceiver", "noDelay", "children"]);

      if (!container) {
        return _react["default"].createElement(_reactAtomicMolecule.SemanticUI, resetProps, children);
      }

      var targetInfo = _objectSpread({}, this.state, {
        targetId: targetId
      });

      var thisChildren = children;

      if (!thisChildren && (0, _react.isValidElement)(container)) {
        thisChildren = container.props.children;
      }

      return (0, _reactAtomicMolecule.build)(container)(_objectSpread({}, resetProps, {
        targetInfo: targetInfo
      }), thisChildren);
    }
  }], [{
    key: "getStores",
    value: function getStores(props) {
      return props.noDelay ? [_fastScrollStore["default"]] : [_scrollStore["default"]];
    }
  }, {
    key: "calculateState",
    value: function calculateState(prevState, props) {
      var noDelay = props.noDelay,
          targetId = props.targetId,
          scrollMargin = props.scrollMargin;
      var store = noDelay ? _fastScrollStore["default"] : _scrollStore["default"];
      var state = store.getState();
      var isShown = prevState && prevState.isShown || false;
      var pos = store.getOffset(targetId, store.storeName) || {};
      var scrollInfo = store.getMap('scroll');
      var scrollTop = scrollInfo.top;

      if (pos.isOnScreen) {
        isShown = true;
      }

      var active = 'undefined' !== typeof targetId && targetId === state.get('m' + scrollMargin);

      if (!isNaN(scrollMargin)) {
        store.addMargin(scrollMargin);
      }

      return _objectSpread({}, pos, {
        active: active,
        scrollTop: scrollTop,
        scrollInfo: scrollInfo,
        isShown: isShown
      });
    }
  }]);
  return ScrollReceiver;
}(_react.Component);

(0, _defineProperty2["default"])(ScrollReceiver, "defaultProps", {
  scrollMargin: 'default',
  isScrollReceiver: true,
  noDelay: false
});

var _default = (0, _reshowFlux.connect)(ScrollReceiver, {
  withProps: true
});

exports["default"] = _default;
module.exports = exports.default;
