// react-atomic-molecule/build/cjs/ui/molecules/Image.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("reshow-runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _classLib = require("class-lib");

var _SemanticUI = _interopRequireDefault(require("../molecules/SemanticUI"));

var Image = function Image(props) {
  var alt = props.alt,
      title = props.title,
      ui = props.ui;
  var classes = (0, _classLib.mixClass)(props.className, {
    image: ui
  });
  var thisAlt = alt;

  if (!thisAlt) {
    thisAlt = title;
  }

  if (props.atom && 'img' !== props.atom) {
    return _react["default"].createElement(_SemanticUI["default"], (0, _extends2["default"])({}, props, {
      className: classes
    }), _react["default"].createElement(_SemanticUI["default"], {
      atom: "img",
      src: props.src,
      alt: thisAlt
    }), props.children);
  } else {
    return _react["default"].createElement(_SemanticUI["default"], (0, _extends2["default"])({
      atom: "img"
    }, props, {
      alt: thisAlt,
      className: classes
    }));
  }
};

Image.defaultProps = {
  ui: true
};
var _default = Image;
exports["default"] = _default;
module.exports = exports.default;
