// react-atomic-molecule/build/cjs/ui/molecules/InputBox.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("reshow-runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("reshow-runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classLib = require("class-lib");

var _SemanticUI = _interopRequireDefault(require("../molecules/SemanticUI"));

var _Button = _interopRequireDefault(require("../molecules/Button"));

var _Label = _interopRequireDefault(require("../molecules/Label"));

var _Icon = _interopRequireDefault(require("../molecules/Icon"));

/* jshint esnext: true */
var InputBox = function InputBox(props) {
  var icon = props.icon,
      button = props.button,
      buttonProps = props.buttonProps,
      className = props.className,
      children = props.children,
      messageType = props.messageType,
      leftLabel = props.leftLabel,
      leftLabelProps = props.leftLabelProps,
      rightLabel = props.rightLabel,
      rightLabelProps = props.rightLabelProps,
      style = props.style,
      transparent = props.transparent,
      inputStyle = props.inputStyle,
      others = (0, _objectWithoutProperties2["default"])(props, ["icon", "button", "buttonProps", "className", "children", "messageType", "leftLabel", "leftLabelProps", "rightLabel", "rightLabelProps", "style", "transparent", "inputStyle"]);
  var classes = (0, _classLib.mixClass)(className, messageType, 'input', {
    labeled: leftLabel || rightLabel,
    right: rightLabel,
    action: button && !icon,
    icon: icon,
    transparent: transparent
  });
  var thisLeftLabel;

  if (leftLabel) {
    thisLeftLabel = _react["default"].createElement(_Label["default"], leftLabelProps, leftLabel);
  }

  var thisRightLabel;

  if (rightLabel) {
    thisRightLabel = _react["default"].createElement(_Label["default"], (0, _extends2["default"])({
      style: Styles.rightLabel,
      className: "basic"
    }, rightLabelProps), rightLabel);
  }

  var thisButton = null;

  if (button) {
    thisButton = _react["default"].createElement(_Button["default"], buttonProps, button);
  }

  if (icon) {
    thisButton = _react["default"].createElement(_Icon["default"], {
      style: Styles.icon
    }, icon);
  }

  return _react["default"].createElement(_SemanticUI["default"], {
    className: classes,
    style: style
  }, thisLeftLabel, _react["default"].createElement(_SemanticUI["default"], (0, _extends2["default"])({
    atom: "input"
  }, others, {
    style: inputStyle,
    ui: false
  })), thisRightLabel, children, thisButton);
};

InputBox.defaultProps = {
  button: 'Submit'
};
var _default = InputBox;
exports["default"] = _default;
var Styles = {
  rightLabel: {
    borderRadius: 0
  },
  icon: {
    width: '1.1em',
    position: 'absolute',
    top: 0,
    right: 0,
    opacity: .5
  }
};
module.exports = exports.default;
