// react-atomic-molecule/build/cjs/ui/molecules/Message.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("reshow-runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("reshow-runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classLib = require("class-lib");

var _Header = _interopRequireDefault(require("../molecules/Header"));

var _SemanticUI = _interopRequireDefault(require("../molecules/SemanticUI"));

var Message = function Message(props) {
  var header = props.header,
      className = props.className,
      children = props.children,
      messageType = props.messageType,
      reset = (0, _objectWithoutProperties2["default"])(props, ["header", "className", "children", "messageType"]);
  var classes = (0, _classLib.mixClass)(className, {
    info: messageType === 'info',
    error: messageType === 'error',
    success: messageType === 'success',
    warning: messageType === 'warning'
  }, 'message');
  var thisHeader;

  if (header) {
    thisHeader = _react["default"].createElement(_Header["default"], null, header);
  }

  return _react["default"].createElement(_SemanticUI["default"], (0, _extends2["default"])({}, reset, {
    className: classes
  }), thisHeader, children);
};

var _default = Message;
exports["default"] = _default;
module.exports = exports.default;
