// react-atomic-molecule/build/cjs/src/lib/styles/index.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ucfirst = _interopRequireDefault(require("ucfirst"));

var _style = _interopRequireDefault(require("./style"));

var _store = _interopRequireDefault(require("./store"));

var _cssNumberToUnit = _interopRequireDefault(require("./cssNumberToUnit"));

var isArray = Array.isArray;
var keys = Object.keys;
var Browser = {
  webkit: 'Webkit',
  ms: 'ms',
  moz: 'Moz'
};

var genStyleId = function genStyleId() {
  _store["default"].counter += 1;
  return 'c' + _store["default"].counter + '_';
};

var createStyle = function createStyle(css, selector, styleId) {
  if (!css) {
    return;
  }

  if ('undefined' === typeof styleId) {
    styleId = genStyleId();
  } else if (_store["default"].registry[styleId]) {
    return _store["default"].registry[styleId];
  }

  if (!isArray(css)) {
    css = [css];
  }

  var styles = [];
  css.forEach(function (one, i) {
    styles[i] = {};
    keys(one).forEach(function (key) {
      if (isArray(one[key]) && 1 === one[key].length) {
        var ucFirstKey = (0, _ucfirst["default"])(key);
        styles[i][Browser.webkit + ucFirstKey] = styles[i][Browser.ms + ucFirstKey] = styles[i][Browser.moz + ucFirstKey] = styles[i][key] = (0, _cssNumberToUnit["default"])(key, one[key][0]);
      } else {
        styles[i][key] = (0, _cssNumberToUnit["default"])(key, one[key]);
      }
    });
  });
  var styleDecl = new _style["default"](styles, selector, styleId);

  _store["default"].newStyles.push(styleDecl);

  return styleDecl;
};

var _default = createStyle; // reactStyle('', '', '')

exports["default"] = _default;
module.exports = exports.default;
