// get-object-value/build/es/src/index.js
import _typeof from "reshow-runtime/es/helpers/typeof";
import { DEFAULT, FUNCTION } from 'reshow-constant';
var isArray = Array.isArray;
var keys = Object.keys;

var getWebpack4Default = function getWebpack4Default(o) {
  return get(o, [DEFAULT, DEFAULT], function () {
    return get(o, [DEFAULT], function () {
      return o;
    });
  });
};

var toJS = function toJS(v) {
  return v && v.toJS ? v.toJS() : v;
};

var toMap = function toMap(a, path) {
  var next = get(a, path, {});
  var nextMap = {};
  keys(next).forEach(function (key) {
    return nextMap[key] = toJS(next[key]);
  });
  return nextMap;
};

var initMap = function initMap(o) {
  return function (k, defaultValue) {
    return o[k] || (o[k] = getDefaultValue(defaultValue));
  };
};

var getDefaultValue = function getDefaultValue(v) {
  return FUNCTION === _typeof(v) ? v() : v;
};

var get = function get(o, path, defaultValue) {
  if (null == o) {
    return getDefaultValue(defaultValue);
  }

  var current = toJS(o);

  if (!path || !isArray(path)) {
    return current;
  }

  path.every(function (a) {
    if (null != current[a]) {
      current = current[a];
      return true;
    } else {
      current = getDefaultValue(defaultValue);
      return false;
    }
  });
  return current;
};

export default get;
export { getWebpack4Default as getDefault, toJS, toMap, initMap };
