// organism-react-scroll-nav/build/es/src/stores/scrollStore.js
import _classCallCheck from "reshow-runtime/es/helpers/classCallCheck";
import _createClass from "reshow-runtime/es/helpers/createClass";
import _possibleConstructorReturn from "reshow-runtime/es/helpers/possibleConstructorReturn";
import _getPrototypeOf from "reshow-runtime/es/helpers/getPrototypeOf";
import _assertThisInitialized from "reshow-runtime/es/helpers/assertThisInitialized";
import _inherits from "reshow-runtime/es/helpers/inherits";
import _defineProperty from "reshow-runtime/es/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import { Set, Map } from 'immutable';
import { ReduceStore } from 'reshow-flux';
import getScrollInfo from 'get-scroll-info';
import { isOnScreen } from 'get-window-offset';
import getOffset from 'getoffset';
import get, { toJS } from 'get-object-value';
import dispatcher, { scrollDispatch } from '../scrollDispatcher';
import testForPassiveScroll from '../testForPassiveScroll';
var incNum = 0;
var DEFAULT_SCROLL_ID = -1;
var keys = Object.keys;

var scrollStore =
/*#__PURE__*/
function (_ReduceStore) {
  _inherits(scrollStore, _ReduceStore);

  function scrollStore() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, scrollStore);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(scrollStore)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "storeName", 'delayScroll');

    _defineProperty(_assertThisInitialized(_this), "isInitEvent", {});

    _defineProperty(_assertThisInitialized(_this), "isInitResizeEvent", false);

    _defineProperty(_assertThisInitialized(_this), "spys", {});

    return _this;
  }

  _createClass(scrollStore, [{
    key: "getInitialState",
    value: function getInitialState() {
      this.trigger = this.triggerScroll.bind(this);
      this.arrNode = Map();
      this.margins = Set();
      this.scrollMonitor = this.runScrollMonitor.bind(this);
      this.bindHandleResize = this.handleResize.bind(this);
      return Map({
        scrollDelay: 50,
        scrollMargin: 0
      });
    }
  }, {
    key: "initResizeEvent",
    value: function initResizeEvent() {
      if ('undefined' !== typeof window) {
        this.isInitResizeEvent = true;
        var win = window;

        if (win.addEventListener) {
          win.addEventListener('resize', this.bindHandleResize);
        } else {
          win.attachEvent('onresize', this.bindHandleResize);
        }
      }
    }
  }, {
    key: "initEvent",
    value: function initEvent(el) {
      if ('undefined' !== typeof el) {
        var self = this;

        if (el.addEventListener) {
          var supportsPassive = testForPassiveScroll();
          el.addEventListener('scroll', self.scrollMonitor, supportsPassive ? {
            passive: true
          } : false);
        } else {
          el.attachEvent('onscroll', self.scrollMonitor);
        }

        setTimeout(function () {
          var fakeEvent = {
            target: el
          };
          self.scrollMonitor(fakeEvent); //for lazy content

          setTimeout(function () {
            return self.scrollMonitor(fakeEvent);
          }, 777);
        });

        if (!self.isInitResizeEvent) {
          self.initResizeEvent();
        }
      }
    }
  }, {
    key: "removeEvent",
    value: function removeEvent(el) {
      el.removeEventListener('scroll', this.scrollMonitor);
    }
  }, {
    key: "handleResize",
    value: function handleResize() {
      var _this2 = this;

      keys(this.spys).forEach(function (scrollId) {
        return _this2.runScrollMonitor({
          target: {
            id: scrollId
          }
        });
      });
    }
  }, {
    key: "runScrollMonitor",
    value: function runScrollMonitor(e) {
      var _this3 = this;

      clearTimeout(this._scrollTimeout);
      var self = this;
      var delay = self.getState().get('scrollDelay');
      self._scrollTimeout = setTimeout(function () {
        return _this3.trigger(e && e.target);
      }, delay);
    }
  }, {
    key: "triggerScroll",
    value: function triggerScroll(scrollNode) {
      var scrollId = get(scrollNode, ['id']) || DEFAULT_SCROLL_ID;
      var defaultMargin = this.getState().get('scrollMargin');
      var actives = {
        mdefault: null
      };
      var offsetCache = {};
      var arrMonitorScroll = [];
      var scroll = getScrollInfo();
      var scrollTop = scroll.top + defaultMargin;
      var margin;
      (this.spys[scrollId] || []).forEach(function (node) {
        var nodeEl = node.getOffsetEl();

        var _get = get(node, ['props'], {}),
            monitorScroll = _get.monitorScroll,
            scrollMargin = _get.scrollMargin;

        var pos = getOffset(nodeEl);

        if (monitorScroll) {
          if (scrollTop >= pos.top && scrollTop < pos.bottom) {
            actives.mdefault = node.id;
          }

          arrMonitorScroll.push(node);
        }

        margin = scrollMargin ? scrollMargin : defaultMargin;
        pos = isOnScreen(pos, scroll, margin);
        offsetCache[node.id] = pos;
      });
      this.margins.forEach(function (margin) {
        scrollTop = scroll.top + margin;
        actives['m' + margin] = null;
        arrMonitorScroll.every(function (node) {
          var pos = offsetCache[node.id];

          if (scrollTop >= pos.top && scrollTop < pos.bottom - 1) {
            actives['m' + margin] = node.id;
            return false;
          }

          return true;
        });
      });
      this.margins = this.margins.clear();
      scrollDispatch(_objectSpread({}, actives, {
        nodes: offsetCache,
        scroll: scroll,
        storeName: this.storeName
      }));
    }
  }, {
    key: "getNode",
    value: function getNode(id) {
      if (this.arrMap && this.arrMap.get) {
        return toJS(this.arrMap.get(id));
      }
    }
  }, {
    key: "getOffset",
    value: function getOffset(id, callName) {
      var nodes = this.getMap('nodes');
      return nodes[id];
    }
  }, {
    key: "hasAttach",
    value: function hasAttach(node) {
      var attachToId = this.getAttachToId(node);

      if (this.spys[attachToId] && this.spys[attachToId].has(node)) {
        return attachToId;
      } else {
        return false;
      }
    }
  }, {
    key: "getNodeId",
    value: function getNodeId(node) {
      if (!node.id) {
        if (node.props && node.props.id) {
          node.id = node.props.id;
        } else {
          node.id = 'spy-' + incNum;
          incNum++;
        }
      }

      return node.id;
    }
  }, {
    key: "getAttachToId",
    value: function getAttachToId(node) {
      var attachTo = get(node, ['props', 'attachTo']);
      var attachToId;

      if (attachTo) {
        node.attachTo = attachTo;
        attachToId = this.getNodeId(attachTo);
      } else {
        if ('undefined' !== typeof window) {
          node.attachTo = window;
        }

        attachToId = DEFAULT_SCROLL_ID;
      }

      return attachToId;
    }
  }, {
    key: "attach",
    value: function attach(node) {
      var nodeId = this.getNodeId(node);
      var attachToId = this.getAttachToId(node);

      if (!this.spys[attachToId]) {
        this.spys[attachToId] = Set().add(node);
      } else {
        this.spys[attachToId] = this.spys[attachToId].add(node);
      }

      this.arrNode = this.arrNode.set(nodeId, node);

      if (!this.isInitEvent[attachToId]) {
        this.isInitEvent[attachToId] = true;
        this.initEvent(node.attachTo);
      }

      return nodeId;
    }
  }, {
    key: "detach",
    value: function detach(node) {
      var attachToId = this.hasAttach(node);

      if (attachToId) {
        this.spys[attachToId] = this.spys[attachToId].remove(node);
        this.arrNode = this.arrNode["delete"](node.id);

        if (!this.spys[attachToId].size) {
          this.removeEvent(node.attachTo);
          delete this.spys[attachToId];
          this.isInitEvent[attachToId] = false;
        }
      }
    }
  }, {
    key: "addMargin",
    value: function addMargin(num) {
      this.margins = this.margins.add(num);
    }
  }, {
    key: "deleteMargin",
    value: function deleteMargin(num) {
      this.margins = this.margins.remove(num);
    }
  }, {
    key: "reduce",
    value: function reduce(state, action) {
      var storeName = get(action, ['storeName'], 'delayScroll');

      if (storeName === this.storeName) {
        return state.merge(action);
      } else {
        return state;
      }
    }
  }]);

  return scrollStore;
}(ReduceStore);

export default new scrollStore(dispatcher);
export { scrollStore };
