// react-atomic-molecule/build/es/src/lib/styles/index.js
import ucfirst from 'ucfirst';
import style from './style';
import store from './store';
import nToU from './cssNumberToUnit';
var isArray = Array.isArray;
var keys = Object.keys;
var Browser = {
  webkit: 'Webkit',
  ms: 'ms',
  moz: 'Moz'
};

var genStyleId = function genStyleId() {
  store.counter += 1;
  return 'c' + store.counter + '_';
};

var createStyle = function createStyle(css, selector, styleId) {
  if (!css) {
    return;
  }

  if ('undefined' === typeof styleId) {
    styleId = genStyleId();
  } else if (store.registry[styleId]) {
    return store.registry[styleId];
  }

  if (!isArray(css)) {
    css = [css];
  }

  var styles = [];
  css.forEach(function (one, i) {
    styles[i] = {};
    keys(one).forEach(function (key) {
      if (isArray(one[key]) && 1 === one[key].length) {
        var ucFirstKey = ucfirst(key);
        styles[i][Browser.webkit + ucFirstKey] = styles[i][Browser.ms + ucFirstKey] = styles[i][Browser.moz + ucFirstKey] = styles[i][key] = nToU(key, one[key][0]);
      } else {
        styles[i][key] = nToU(key, one[key]);
      }
    });
  });
  var styleDecl = new style(styles, selector, styleId);
  store.newStyles.push(styleDecl);
  return styleDecl;
};

export default createStyle; // reactStyle('', '', '')
