// react-atomic-molecule/build/es/src/lib/styles/injectStyle.js
import _objectSpread from "reshow-runtime/es/helpers/objectSpread2";
import { doc } from 'win-doc';
import store from './store.js';
import stylesToCSS from './stylesToCSS';
import { inject, create } from 'create-el';
import query from 'css-query-selector';

var reInjectStyle = function reInjectStyle() {
  store.newStyles = Object.values(store.registry);
  injectStyle();
};

var keys = Object.keys;

var appendCss = function appendCss(cssArr) {
  keys(cssArr).forEach(function (key) {
    var id = 'react-style-' + key;
    var styleDom = query.one('#' + id);
    var css = cssArr[key];

    if (styleDom) {
      styleDom.innerHTML = css;
    } else {
      styleDom = create('style')()({
        id: id,
        innerHTML: css
      });
      inject(function () {
        return doc().getElementsByTagName('head')[0];
      })(styleDom);
    }
  });
};

var injectStyle = function injectStyle() {
  if (!store.newStyles.length) {
    // We are in Node or Styles are already injected
    return null;
  }

  var compiled = stylesToCSS(store.newStyles);
  store.newStyles = [];
  store.registry = _objectSpread({}, store.registry, {}, compiled.styleIds);

  if (compiled.css) {
    if (doc().URL) {
      appendCss(compiled.cssArr);
    } else {
      console.log(compiled.css);
    }
  }
};

export { reInjectStyle };
export default injectStyle;
