// get-object-value/build/cjs/src/index.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.initMap = exports.toMap = exports.toJS = exports.getDefault = exports["default"] = void 0;

var _typeof2 = _interopRequireDefault(require("reshow-runtime/helpers/typeof"));

var _reshowConstant = require("reshow-constant");

var isArray = Array.isArray;
var keys = Object.keys;

var getWebpack4Default = function getWebpack4Default(o) {
  return get(o, [_reshowConstant.DEFAULT, _reshowConstant.DEFAULT], function () {
    return get(o, [_reshowConstant.DEFAULT], function () {
      return o;
    });
  });
};

exports.getDefault = getWebpack4Default;

var toJS = function toJS(v) {
  return v && v.toJS ? v.toJS() : v;
};

exports.toJS = toJS;

var toMap = function toMap(a, path) {
  var next = get(a, path, {});
  var nextMap = {};
  keys(next).forEach(function (key) {
    return nextMap[key] = toJS(next[key]);
  });
  return nextMap;
};

exports.toMap = toMap;

var initMap = function initMap(o) {
  return function (k, defaultValue) {
    return o[k] || (o[k] = getDefaultValue(defaultValue));
  };
};

exports.initMap = initMap;

var getDefaultValue = function getDefaultValue(v) {
  return _reshowConstant.FUNCTION === (0, _typeof2["default"])(v) ? v() : v;
};

var get = function get(o, path, defaultValue) {
  if (null == o) {
    return getDefaultValue(defaultValue);
  }

  var current = toJS(o);

  if (!path || !isArray(path)) {
    return current;
  }

  path.every(function (a) {
    if (null != current[a]) {
      current = current[a];
      return true;
    } else {
      current = getDefaultValue(defaultValue);
      return false;
    }
  });
  return current;
};

var _default = get;
exports["default"] = _default;
