// css-query-selector/build/cjs/src/queryFrom.js
"use strict";

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultQuery = exports["default"] = void 0;

var _typeof2 = _interopRequireDefault(require("reshow-runtime/helpers/typeof"));

var _toConsumableArray2 = _interopRequireDefault(require("reshow-runtime/helpers/toConsumableArray"));

var _winDoc = require("win-doc");

var _reshowConstant = require("reshow-constant");

var arrayFrom = function arrayFrom(a) {
  return (0, _toConsumableArray2["default"])(a);
};

var findHit = function findHit(all, el) {
  var hit;

  var setHit = function setHit(p) {
    return hit = p;
  };

  all.some(function (p) {
    return p.contains(el) && !p.isSameNode(el) ? setHit(p) : false;
  });
  return hit;
};

var queryFrom = function queryFrom(base) {
  var myBase = _reshowConstant.FUNCTION === (0, _typeof2["default"])(base) ? base : function () {
    return defaultQuery.el(base);
  };

  var queryOne = function queryOne(sel) {
    var _myBase;

    return (_myBase = myBase()) === null || _myBase === void 0 ? void 0 : _myBase.querySelector(sel);
  };

  var queryAll = function queryAll(sel) {
    var _myBase2;

    return arrayFrom((_myBase2 = myBase()) === null || _myBase2 === void 0 ? void 0 : _myBase2.querySelectorAll(sel));
  };

  var queryEl = function queryEl(el) {
    return _reshowConstant.STRING === (0, _typeof2["default"])(el) ? queryOne(el) : el;
  };

  var _queryAncestorPolyfill = function _queryAncestorPolyfill(el, ancestor) {
    var lastHit;
    var hit;
    var all = queryAll(ancestor);

    if (all) {
      hit = findHit(all, el);
    }

    while (hit) {
      lastHit = hit;
      all = hit.querySelectorAll(ancestor);

      if (all) {
        hit = findHit(arrayFrom(all), el);
      } else {
        break;
      }
    }

    return lastHit;
  };

  var queryAncestor = function queryAncestor(el, ancestor) {
    el = queryEl(el);
    return el.closest ? el.closest(ancestor) : _queryAncestorPolyfill(el, ancestor);
  };

  return {
    all: queryAll,
    ancestor: queryAncestor,
    el: queryEl,
    one: queryOne
  };
};

var defaultQuery = queryFrom(_winDoc.doc);
exports.defaultQuery = defaultQuery;
var _default = queryFrom;
exports["default"] = _default;
