// organism-react-scroll-nav/build/cjs/ui/pages/index.js
"use strict";

var _interopRequireWildcard = require("reshow-runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("reshow-runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("reshow-runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("reshow-runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("reshow-runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("reshow-runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("reshow-runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("reshow-runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("reshow-runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("reshow-runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _index = require("../../src/index");

var _reactAtomicMolecule = require("react-atomic-molecule");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

(0, _index.scrollDispatch)({
  scrollMargin: 100
});

var MenuItem =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2["default"])(MenuItem, _Component);

  function MenuItem() {
    (0, _classCallCheck2["default"])(this, MenuItem);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(MenuItem).apply(this, arguments));
  }

  (0, _createClass2["default"])(MenuItem, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          targetInfo = _this$props.targetInfo,
          style = _this$props.style,
          reset = (0, _objectWithoutProperties2["default"])(_this$props, ["targetInfo", "style"]);
      var activeStyle = null;

      if (targetInfo.active) {
        activeStyle = Styles.active;
      }

      return _react["default"].createElement("div", (0, _extends2["default"])({
        style: _objectSpread({}, Styles.link, {}, style, {}, activeStyle)
      }, reset));
    }
  }]);
  return MenuItem;
}(_react.Component);

var _ref =
/*#__PURE__*/
_react["default"].createElement(MenuItem, null);

var _ref2 =
/*#__PURE__*/
_react["default"].createElement(MenuItem, null);

var _ref3 =
/*#__PURE__*/
_react["default"].createElement(MenuItem, null);

var Index =
/*#__PURE__*/
function (_Component2) {
  (0, _inherits2["default"])(Index, _Component2);

  function Index() {
    (0, _classCallCheck2["default"])(this, Index);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(Index).apply(this, arguments));
  }

  (0, _createClass2["default"])(Index, [{
    key: "render",
    value: function render() {
      return _react["default"].createElement("div", null, _react["default"].createElement("div", {
        id: "nav-parent",
        style: Styles.nav
      }, _react["default"].createElement(_index.SmoothScrollLink, {
        scrollRefId: "nav-parent",
        container: _ref,
        targetId: "t1",
        style: {
          background: '#3498DB'
        }
      }, "test1"), _react["default"].createElement(_index.SmoothScrollLink, {
        scrollRefId: "nav-parent",
        container: _ref2,
        targetId: "t2",
        style: {
          background: '#F1C40F'
        }
      }, "test2"), _react["default"].createElement(_index.SmoothScrollLink, {
        scrollRefId: "nav-parent",
        container: _ref3,
        targetId: "t3",
        style: {
          background: '#2ECC71'
        }
      }, "test3")), _react["default"].createElement("div", {
        style: {
          paddingTop: 66
        }
      }, _react["default"].createElement(_index.ScrollSpy, {
        id: "t1",
        style: _objectSpread({}, Styles.content, {
          background: "#3498DB"
        })
      }, "test111"), _react["default"].createElement(_index.ScrollSpy, {
        id: "t2",
        style: _objectSpread({}, Styles.content, {
          background: "#F1C40F"
        })
      }, _react["default"].createElement(_index.ScrollSpy, {
        monitorScroll: false
      }, _react["default"].createElement(_index.ScrollReceiver, {
        style: {
          border: "1px solid #fff"
        }
      }, "test222"))), _react["default"].createElement(_index.ScrollSpy, {
        id: "t3",
        style: _objectSpread({}, Styles.content, {
          background: "#2ECC71"
        })
      }, "test333")));
    }
  }]);
  return Index;
}(_react.Component);

var _default = Index;
exports["default"] = _default;
var Styles = {
  content: {
    height: "500px",
    textAlign: "center",
    paddingTop: "10%",
    fontSize: "4rem",
    color: "#fff"
  },
  link: {
    display: "inline-block",
    padding: "0.3rem 0.5rem",
    fontSize: "2rem",
    color: "#fff",
    marginRight: "10px",
    opacity: "0.5",
    cursor: "pointer"
  },
  active: {
    opacity: 1
  },
  body: (0, _reactAtomicMolecule.reactStyle)({
    margin: 0,
    padding: 0
  }, 'body'),
  nav: {
    position: 'fixed',
    background: '#fff',
    textAlign: 'center',
    width: "100%",
    padding: "10px 0",
    top: 0
  }
};
(0, _reactAtomicMolecule.injectStyle)();
module.exports = exports.default;
