// reshow-flux/build/es/connectHook.js
import _objectSpread from "reshow-runtime/es/helpers/objectSpread2";
import _slicedToArray from "reshow-runtime/es/helpers/slicedToArray";
import { useMemo, useState, useEffect } from 'react';
import build from 'reshow-build';
import dedup from 'array.dedup';
import { CHANGE } from 'reshow-flux-base';

var connectHook = function connectHook(Base, options) {
  var _ref = options || {},
      getStores = _ref.getStores,
      calculateState = _ref.calculateState,
      defaultProps = _ref.defaultProps,
      displayName = _ref.displayName;

  var Connected = function Connected(props) {
    var _useState = useState(function () {
      return {
        state: calculateState({}, props),
        props: props
      };
    }),
        _useState2 = _slicedToArray(_useState, 2),
        data = _useState2[0],
        setData = _useState2[1];

    useEffect(function () {
      var stores = dedup(getStores(props)) || [];

      if (stores && stores.length) {
        var __unmount__;

        var handleChange = function handleChange() {
          if (!__unmount__) {
            setData(function (prev) {
              return {
                __init__: true,
                props: props,
                state: _objectSpread({}, prev.state, {}, props, {}, calculateState(prev.state, props))
              };
            });
          }
        };

        stores.forEach(function (store) {
          return store.addListener(handleChange, CHANGE);
        });

        if (!data.__init__ || data.props !== props) {
          handleChange();
        }

        return function () {
          __unmount__ = true;
          stores.forEach(function (store) {
            return store.removeListener(handleChange, CHANGE);
          });
        };
      }
    }, [props]);
    return useMemo(function () {
      return build(Base)(data.state);
    }, [data.state]);
  };

  var componentName = displayName || Base.displayName || Base.name;
  Connected.displayName = 'HookConnected(' + componentName + ')';

  if (defaultProps) {
    Connected.defaultProps = defaultProps;
  }

  return Connected;
};

export default connectHook;
